File: app/src/main/java/com/example/finalproject/model/Recipe.java
package com.example.finalproject.model;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.SerializedName;

public class Recipe implements Parcelable {

    @SerializedName("uri")
    public String id;

    @SerializedName("label")
    public String title;

    @SerializedName("image")
    public String img;

    public Recipe(String id, String title){
        this.id = id;
        this.title = title;
    }

    protected Recipe(Parcel in) {
        this(in.readString() ,in.readString());
    }

    public static final Creator<Recipe> CREATOR = new Creator<Recipe>() {
        @Override
        public Recipe createFromParcel(Parcel in) {
            return new Recipe(in);
        }

        @Override
        public Recipe[] newArray(int size) {
            return new Recipe[size];
        }
    };

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public String getImg() {
        return img;
    }
    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int i) {
        dest.writeString(id);
        dest.writeString(title);
        dest.writeString(img);
    }
}

File: app/src/main/java/com/example/finalproject/model/Recipe.kt
package com.example.finalproject.model

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

class Recipe : Parcelable {
    @SerializedName("uri")
    var id: String?

    @JvmField
    @SerializedName("label")
    var title: String?

    @JvmField
    @SerializedName("image")
    var img: String? = null

    @SerializedName("calories")
    var calories: String?

    @JvmField
    @SerializedName("ingredientLines")
    var ingredients: ArrayList<String>

    @JvmField
    @SerializedName("source")
    var takenFrom: String?

    @JvmField
    @SerializedName("cautions")
    var cautions: ArrayList<String>? = null

    @JvmField
    @SerializedName("healthLabels")
    var healthLabels: ArrayList<String>? = null


    constructor(
        id: String?,
        title: String?,
        calories: String?,
        ingredients: ArrayList<String>,
        takenFrom: String?,
        cautions: ArrayList<String>?,
        healthLabels: ArrayList<String>?,
    ) {
        this.id = id
        this.title = title
        this.calories = calories
        this.ingredients = ingredients
        this.takenFrom = takenFrom
        this.cautions = cautions
        this.healthLabels = healthLabels
    }

    protected constructor(`in`: Parcel) {
        id = `in`.readString()
        title = `in`.readString()
        calories = `in`.readString()
        ingredients = ArrayList()
        `in`.readStringList(ingredients)
        takenFrom = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, i: Int) {
        dest.writeString(id)
        dest.writeString(title)
        dest.writeString(img)
        dest.writeString(calories)
        dest.writeStringList(ingredients)
        dest.writeString(takenFrom)
        dest.writeStringList(cautions)
        dest.writeStringList(healthLabels)
    }

//    companion object CREATOR : Creator<Recipe> {
//        override fun createFromParcel(parcel: Parcel): Recipe {
//            return Recipe(parcel)
//        }
//
//        override fun newArray(size: Int): Array<Recipe?> {
//            return arrayOfNulls(size)
//        }
//    }
    companion object CREATOR : Creator<Recipe> {
        override fun createFromParcel(parcel: Parcel): Recipe {
            return Recipe(parcel)
        }

        override fun newArray(size: Int): Array<Recipe?> {
            return arrayOfNulls(size)
        }
    }

}
